packages/core-mobile/app/new/routes/(signedIn)/(tabs)/portfolio/assets.tsx

```diff
@@ -1,27 +1,50 @@
 import React from 'react'
-import { View, Text, ScrollView } from '@avalabs/k2-alpine'
-import {} from 'react-native-gesture-handler'
-import BlurredBarsContentLayout from 'common/components/BlurredBarsContentLayout'
+import { EmptyAssets } from 'features/portfolio/components/assets/EmptyAssets'
+import { useSelector } from 'react-redux'
+import {
+  selectIsAllBalancesInaccurate,
+  selectIsLoadingBalances,
+  selectIsRefetchingBalances,
+  selectTokensWithBalanceForAccount
+} from 'store/balance'
+import { selectActiveAccount } from 'store/account'
+import { RootState } from 'store'
+import { TokensList } from 'features/portfolio/components/assets/TokensList'
+import { TokenType } from '@avalabs/vm-module-types'
+import { ErrorState } from 'features/portfolio/components/assets/ErrorState'
+import { useSearchableTokenList } from 'screens/portfolio/useSearchableTokenList'
+import { LoadingState } from 'features/portfolio/components/assets/LoadingState'
 
-const PortfolioAssetsScreen = (): JSX.Element => {
-  return (
-    <BlurredBarsContentLayout>
-      <ScrollView
-        sx={{
-          flex: 1
-        }}
-        contentContainerSx={{
-          paddingTop: 16,
-          flex: 1,
-          alignItems: 'center',
-          gap: 16
-        }}>
-        <View>
-          <Text variant="heading3">Assets</Text>
-        </View>
-      </ScrollView>
-    </BlurredBarsContentLayout>
+const PortfolioAssetsScreen = (): JSX.Element | undefined => {
+  const { refetch } = useSearchableTokenList()
+  const activeAccount = useSelector(selectActiveAccount)
+  const tokens = useSelector((state: RootState) =>
+    selectTokensWithBalanceForAccount(state, activeAccount?.index)
   )
+  const isAllBalancesInaccurate = useSelector(
+    selectIsAllBalancesInaccurate(activeAccount?.index ?? 0)
+  )
+
+  const nonNftTokens = tokens.filter(
+    token => token.type !== TokenType.ERC1155 && token.type !== TokenType.ERC721
+  )
+  const isBalanceLoading = useSelector(selectIsLoadingBalances)
+  const isRefetchingBalance = useSelector(selectIsRefetchingBalances)
+
+  const renderContent = (): React.JSX.Element => {
+    if (isBalanceLoading || isRefetchingBalance) {
+      return <LoadingState />
+    }
+    if (isAllBalancesInaccurate) {
```

Comment:
the inaccurate state actually has a different UX. the error state should be used for network errors.